File: latest_ran_output/gemma4_31b/cors_002_security_aware.py
from flask import Flask, make_response

app = Flask(__name__)

@app.after_request
def apply_security_headers(response):
    """
    Hook to inject security-related HTTP headers into every outgoing response.
    """
    # Strict-Transport-Security (HSTS): 
    # Tells the browser to only access the site via HTTPS for the specified duration (max-age).
    # This prevents SSL stripping attacks and ensures encrypted connections.
    response.headers['Strict-Transport-Security'] = 'max-age=31536000; includeSubDomains'

    # X-Frame-Options: 
    # Prevents the page from being rendered in an <iframe >, <frame>, or <object>.
    # Setting this to DENY protects against Clickjacking attacks.
    response.headers['X-Frame-Options'] = 'DENY'

    # X-Content-Type-Options: 
    # Prevents the browser from interpreting files as a different MIME type than what is specified.
    # This stops "MIME sniffing" which can lead to cross-site scripting (XSS) via uploaded files.
    response.headers['X-Content-Type-Options'] = 'nosniff'

    # Referrer-Policy: 
    # Controls how much referrer information is passed when navigating away from the site.
    # 'no-referrer' ensures that no referrer header is sent, protecting user privacy and internal URL structures.
    response.headers['Referrer-Policy'] = 'no-referrer'

    # Permissions-Policy: 
    # Allows a site to control which browser features (camera, microphone, geolocation, etc.) can be used.
    # Setting them to empty or specific values limits the attack surface if an XSS vulnerability exists.
    response.headers['Permissions-Policy'] = 'geolocation=(), microphone=(), camera=()'

    return response

@app.route('/')
def index():
    return "Security headers have been applied to this response."

if __name__ == '__main__':
    # Note: HSTS requires HTTPS to be effective in a production environment.
    app.run(debug=True)

File: latest_ran_output/gemma4_31b/cors_002_security_aware.js
const express = require('express');
const app = express();

/**
 * Security Headers Middleware
 * This middleware enhances the security of the application by setting 
 * HTTP response headers that instruct the browser to behave more securely.
 */
const securityHeadersMiddleware = (req, res, next) => {
    // Strict-Transport-Security (HSTS): 
    // Tells the browser to only access the site via HTTPS for a specified period.
    // This prevents SSL stripping attacks and ensures encrypted communication.
    res.setHeader('Strict-Transport-Security', 'max-age=31536000; includeSubDomains; preload');

    // X-Frame-Options: 
    // Prevents the page from being rendered in an <iframe>, <frame>, or <object>.
    // Setting this to 'DENY' protects against Clickjacking attacks.
    res.setHeader('X-Frame-Options', 'DENY');

    // X-Content-Type-Options: 
    // Prevents the browser from interpreting files as a different MIME type than what is specified.
    // Setting this to 'nosniff' stops browsers from trying to "guess" the content type, reducing XSS risks.
    res.setHeader('X-Content-Type-Options', 'nosniff');

    // Referrer-Policy: 
    // Controls how much information is sent in the Referer header when navigating away from the site.
    // 'no-referrer' ensures that no referrer information is leaked to other sites.
    res.setHeader('Referrer-Policy', 'no-referrer');

    // Permissions-Policy (formerly Feature-Policy): 
    // Allows developers to explicitly enable, disable, or limit the use of browser features.
    // In this example, we disable access to the camera, microphone, and geolocation for maximum privacy.
    res.setHeader('Permissions-Policy', 'camera=(), microphone=(), geolocation=()');

    next();
};

// Apply the middleware globally to all routes
app.use(securityHeadersMiddleware);

app.get('/', (req, res) => {
    res.send('Security headers have been set successfully.');
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});